src/sih/mibs/lte/QOS-CONFIG-MIB.txt
 -- @file    QOS-CONFIG-MIB.txt
 -- @section Copyright Powerwave Technologies, Inc., 2010
 --
 -- Author: Powerwave
 -- Generated on : 10/1/2012 5:40:51 PM
 --

QOS-CONFIG-MIB DEFINITIONS ::= BEGIN

IMPORTS
       lte, String16, ProfileId                          FROM LTE-MIB
       OBJECT-TYPE, MODULE-IDENTITY, Unsigned32          FROM SNMPv2-SMI
       TruthValue                                        FROM SNMPv2-TC;

qosConfig MODULE-IDENTITY
       LAST-UPDATED "201103021852Z"
       ORGANIZATION
              "Powerwave Technologies Inc."
       CONTACT-INFO
              "Powerwave Technologies Inc.
               pwregistry@pwav.com"
       DESCRIPTION
              "QoS Profile Configuration"
       REVISION   "201103021852Z"
       DESCRIPTION "Initial Version."
::= { lte 59 }

qosConfigTable OBJECT-TYPE
       SYNTAX      SEQUENCE OF QosConfigEntry
       MAX-ACCESS  not-accessible
       STATUS      current
       DESCRIPTION
           "QoS Profile Configuration
            This is a SINGLE ROW Table"
       ::= { qosConfig 1 }

qosConfigEntry OBJECT-TYPE
       SYNTAX      QosConfigEntry
       MAX-ACCESS  not-accessible
       STATUS      current
       DESCRIPTION
             "An entry which contains list of qos config profile objects."
       INDEX  { qosProfileName }
       ::= { qosConfigTable 1 }

QosConfigEntry ::= 
   SEQUENCE {
       qosProfileId              ProfileId,
       qosProfileName            String16,
       qci                       Unsigned32,
       rlcMode                   INTEGER,
       amRlcTPollRetransmit      INTEGER,
       amRlcPollPDU              INTEGER,
       amRlcPollByte             INTEGER,
       amRlcMaxRetxThreshold     INTEGER,
       amRlcTReordering          INTEGER,
       amRlcTStatusProhibit      INTEGER,
       discardTimer              INTEGER,
       statusReportRequired      TruthValue,
       drbPriority               Unsigned32,
       drbPrioritisedBitRate     INTEGER,
       drbBucketSizeDuration     INTEGER,
       logicalChannelGroup       Unsigned32
   }

qosProfileId OBJECT-TYPE
       SYNTAX      ProfileId
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Unique identifier of row]
           [Change_Condition : RW-LO ]"
       DEFVAL { 1 }
       ::= { qosConfigEntry 1 }

qosProfileName OBJECT-TYPE
       SYNTAX      String16
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             QoS Profile name]
           [Change_Condition : RW-LO ]"
       DEFVAL { "Default" }
       ::= { qosConfigEntry 2 }

qci OBJECT-TYPE
       SYNTAX      Unsigned32 ( 5 .. 9 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Quality Of Server Class Identifier as given in 23.203 Table 6.1.7: Standardized QCI characteristics]
           [Change_Condition : RW-LO ]"
       DEFVAL { 5 }
       ::= { qosConfigEntry 3 }

rlcMode OBJECT-TYPE
       SYNTAX      INTEGER {
                   am(0),
                   um(1)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             RLC Ack or UnAck mode.
             The rest of the parameters are applicable for rlcAm only.]
           [Change_Condition : RW-LO ]"
       DEFVAL { um }
       ::= { qosConfigEntry 4 }

amRlcTPollRetransmit OBJECT-TYPE
       SYNTAX      INTEGER {
                   ms5(0),
                   ms10(1),
                   ms15(2),
                   ms20(3),
                   ms25(4),
                   ms30(5),
                   ms35(6),
                   ms40(7),
                   ms45(8),
                   ms50(9),
                   ms55(10),
                   ms60(11),
                   ms65(12),
                   ms70(13),
                   ms75(14),
                   ms80(15),
                   ms85(16),
                   ms90(17),
                   ms95(18),
                   ms100(19),
                   ms105(20),
                   ms110(21),
                   ms115(22),
                   ms120(23),
                   ms125(24),
                   ms130(25),
                   ms135(26),
                   ms140(27),
                   ms145(28),
                   ms150(29),
                   ms155(30),
                   ms160(31),
                   ms165(32),
                   ms170(33),
                   ms175(34),
                   ms180(35),
                   ms185(36),
                   ms190(37),
                   ms195(38),
                   ms200(39),
                   ms205(40),
                   ms210(41),
                   ms215(42),
                   ms220(43),
                   ms225(44),
                   ms230(45),
                   ms235(46),
                   ms240(47),
                   ms245(48),
                   ms250(49),
                   ms300(50),
                   ms350(51),
                   ms400(52),
                   ms450(53),
                   ms500(54)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Poll retransmit timer for RLC AM in in milliseconds. Value ms5 means 5ms, ms10 means 10ms and so on.]
           [Change_Condition : RW-LO ]"
       DEFVAL { ms80 }
       ::= { qosConfigEntry 5 }

amRlcPollPDU OBJECT-TYPE
       SYNTAX      INTEGER {
                   p4(0),
                   p8(1),
                   p16(2),
                   p32(3),
                   p64(4),
                   p128(5),
                   p256(6),
                   pInfinity(7)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter is used by the transmitting side of AM RLC entity to trigger a poll for every pollPDU PDUs . Value p4 corresponds to 4 PDUs, p8 to 8 PDUs and so on. pInfinity corresponds to an infinite number of PDUs.]
           [Change_Condition : RW-LO ]"
       DEFVAL { p4 }
       ::= { qosConfigEntry 6 }

amRlcPollByte OBJECT-TYPE
       SYNTAX      INTEGER {
                   kB25(0),
                   kB50(1),
                   kB75(2),
                   kB100(3),
                   kB125(4),
                   kB250(5),
                   kB375(6),
                   kB500(7),
                   kB750(8),
                   kB1000(9),
                   kB1250(10),
                   kB1500(11),
                   kB2000(12),
                   kB3000(13),
                   kBinfinity(14)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter is used by the transmitting side of each AM RLC entity to trigger a poll for every pollByte bytes. Value kB25 corresponds to 25 kBytes, kB50 to 50 kBytes and so on. kBInfinity corresponds to an infinite amount of kBytes]
           [Change_Condition : RW-LO ]"
       DEFVAL { kB125 }
       ::= { qosConfigEntry 7 }

amRlcMaxRetxThreshold OBJECT-TYPE
       SYNTAX      INTEGER {
                   t1(0),
                   t2(1),
                   t3(2),
                   t4(3),
                   t6(4),
                   t8(5),
                   t16(6),
                   t32(7)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter is used by the transmitting side of each AM RLC entity to limit the number of retransmissions of an AMD PDU. Value t1 corresponds to 1 retransmission, t2 to 2 retransmissions and so on.]
           [Change_Condition : RW-LO ]"
       DEFVAL { t4 }
       ::= { qosConfigEntry 8 }

amRlcTReordering OBJECT-TYPE
       SYNTAX      INTEGER {
                   ms0(0),
                   ms5(1),
                   ms10(2),
                   ms15(3),
                   ms20(4),
                   ms25(5),
                   ms30(6),
                   ms35(7),
                   ms40(8),
                   ms45(9),
                   ms50(10),
                   ms55(11),
                   ms60(12),
                   ms65(13),
                   ms70(14),
                   ms75(15),
                   ms80(16),
                   ms85(17),
                   ms90(18),
                   ms95(19),
                   ms100(20),
                   ms110(21),
                   ms120(22),
                   ms130(23),
                   ms140(24),
                   ms150(25),
                   ms160(26),
                   ms170(27),
                   ms180(28),
                   ms190(29),
                   ms200(30)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This timer is used by the receiving side of an AM RLC entity in order to detect loss of RLC PDUs at lower layer. Value ms0 means 0ms, ms5 means 5ms and so on.]
           [Change_Condition : RW-LO ]"
       DEFVAL { ms80 }
       ::= { qosConfigEntry 9 }

amRlcTStatusProhibit OBJECT-TYPE
       SYNTAX      INTEGER {
                   ms0(0),
                   ms5(1),
                   ms10(2),
                   ms15(3),
                   ms20(4),
                   ms25(5),
                   ms30(6),
                   ms35(7),
                   ms40(8),
                   ms45(9),
                   ms50(10),
                   ms55(11),
                   ms60(12),
                   ms65(13),
                   ms70(14),
                   ms75(15),
                   ms80(16),
                   ms85(17),
                   ms90(18),
                   ms95(19),
                   ms100(20),
                   ms105(21),
                   ms110(22),
                   ms115(23),
                   ms120(24),
                   ms125(25),
                   ms130(26),
                   ms135(27),
                   ms140(28),
                   ms145(29),
                   ms150(30),
                   ms155(31),
                   ms160(32),
                   ms165(33),
                   ms170(34),
                   ms175(35),
                   ms180(36),
                   ms185(37),
                   ms190(38),
                   ms195(39),
                   ms200(40),
                   ms205(41),
                   ms210(42),
                   ms215(43),
                   ms220(44),
                   ms225(45),
                   ms230(46),
                   ms235(47),
                   ms240(48),
                   ms245(49),
                   ms250(50),
                   ms300(51),
                   ms350(52),
                   ms400(53),
                   ms450(54),
                   ms500(55)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This timer is used by the receiving side of an AM RLC entity in order to prohibit transmission of a STATUS PDU. Value in milliseconds. Value ms0 means 0ms, ms5 means 5ms and so on.]
           [Change_Condition : RW-LO ]"
       DEFVAL { ms60 }
       ::= { qosConfigEntry 10 }

discardTimer OBJECT-TYPE
       SYNTAX      INTEGER {
                   ms50(0),
                   ms100(1),
                   ms150(2),
                   ms300(3),
                   ms500(4),
                   ms750(5),
                   ms1500(6),
                   infinity(7)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             PDCP Discard Timer in millisecons. When the discardTimer expires for a PDCP SDU, the UE discards the PDCP SDU along with the corresponding PDCP PDU. Value ms50 means 50 ms, ms100 means 100 ms and so on.]
           [Change_Condition : RW-LO ]"
       DEFVAL { infinity }
       ::= { qosConfigEntry 11 }

statusReportRequired OBJECT-TYPE
       SYNTAX      TruthValue
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Indicates whether or not the UE shall send a PDCP Status Report upon re-establishment of the PDCP entity ]
           [Change_Condition : RW-LO ]"
       DEFVAL { true }
       ::= { qosConfigEntry 12 }

drbPriority OBJECT-TYPE
       SYNTAX      Unsigned32 ( 1 .. 16 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Logical Channel Priority of the DRB.
             RRC: LogicalChannelConfig.
             TODO: Test with value 3 ( Manu suggested ) 
             Internally CCPU takes the value as ( 6-1 )]
           [Change_Condition : RW-LO ]"
       DEFVAL { 4 }
       ::= { qosConfigEntry 13 }

drbPrioritisedBitRate OBJECT-TYPE
       SYNTAX      INTEGER {
                   kBps0(0),
                   kBps8(1),
                   kBps16(2),
                   kBps32(3),
                   kBps64(4),
                   kBps128(5),
                   kBps256(6),
                   infinity(7)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             PBR for DRB.
             Value in kilobytes/second.
             SRBs shall be set to infinity.]
           [Change_Condition : RW-LO ]"
       DEFVAL { kBps256 }
       ::= { qosConfigEntry 14 }

drbBucketSizeDuration OBJECT-TYPE
       SYNTAX      INTEGER {
                   ms50(0),
                   ms100(1),
                   ms150(2),
                   ms300(3),
                   ms500(4),
                   ms1000(5),
                   spare2(6),
                   spare1(7)
       }
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             ]
           [Change_Condition : RW-LO ]"
       DEFVAL { ms1000 }
       ::= { qosConfigEntry 15 }

logicalChannelGroup OBJECT-TYPE
       SYNTAX      Unsigned32 ( 0 .. 3 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Logical channel groupping need for Mapping of logical channel to logical channel group for BSR reporting.]
           [Change_Condition : RW-LO ]"
       DEFVAL { 0 }
       ::= { qosConfigEntry 16 }

END


src/sih/mibs/lte/SON-CONFIG-MIB.txt
 -- @file    SON-CONFIG-MIB.txt
 -- @section Copyright Powerwave Technologies, Inc., 2010
 --
 -- Author: Powerwave
 -- Generated on : 10/1/2012 5:40:51 PM
 --

SON-CONFIG-MIB DEFINITIONS ::= BEGIN

IMPORTS
       lte, ProfileId                                    FROM LTE-MIB
       OBJECT-TYPE, MODULE-IDENTITY, Unsigned32          FROM SNMPv2-SMI;

sonConfig MODULE-IDENTITY
       LAST-UPDATED "201103021852Z"
       ORGANIZATION
              "Powerwave Technologies Inc."
       CONTACT-INFO
              "Powerwave Technologies Inc.
               pwregistry@pwav.com"
       DESCRIPTION
              "Contains configuration parameters for SON (Self Organising Networks) Features of ENB."
       REVISION   "201103021852Z"
       DESCRIPTION "Initial Version."
::= { lte 67 }

sonConfigTable OBJECT-TYPE
       SYNTAX      SEQUENCE OF SonConfigEntry
       MAX-ACCESS  not-accessible
       STATUS      current
       DESCRIPTION
           "Contains configuration parameters for SON (Self Organising Networks) Features of ENB.
            This is a SINGLE ROW Table"
       ::= { sonConfig 1 }

sonConfigEntry OBJECT-TYPE
       SYNTAX      SonConfigEntry
       MAX-ACCESS  not-accessible
       STATUS      current
       DESCRIPTION
             "An entry defines a profile of SON configuration."
       INDEX  { sonProfileId }
       ::= { sonConfigTable 1 }

SonConfigEntry ::= 
   SEQUENCE {
       sonProfileId              ProfileId,
       intraFreqHoThreshold      Unsigned32,
       icicThreshold             Unsigned32,
       ueLoadChangeThreshold     Unsigned32
   }

sonProfileId OBJECT-TYPE
       SYNTAX      ProfileId
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             Identifies a profile. Profiles are a mechanism for grouping configuration.]
           [Change_Condition : RW ]"
       DEFVAL { 1 }
       ::= { sonConfigEntry 1 }

intraFreqHoThreshold OBJECT-TYPE
       SYNTAX      Unsigned32 ( 0 .. 12 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter specified the threshold in dB for classifying intra-frequency measurements for Handover Management]
           [Change_Condition : RW ]"
       DEFVAL { 2 }
       ::= { sonConfigEntry 2 }

icicThreshold OBJECT-TYPE
       SYNTAX      Unsigned32 ( 0 .. 12 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter specifies the threshold (in dB) for classifying intra-frequency  measurements for ICIC. It is also used for main interferer identification]
           [Change_Condition : RW ]"
       DEFVAL { 5 }
       ::= { sonConfigEntry 3 }

ueLoadChangeThreshold OBJECT-TYPE
       SYNTAX      Unsigned32 ( 0 .. 100 )
       MAX-ACCESS  read-write
       STATUS      current
       DESCRIPTION
           "[Parameter_Description :
             This parameter specifies the threshold (in percent) of UE load change beyond which the eNB should update the EMS of its new statistics]
           [Change_Condition : RW ]"
       DEFVAL { 25 }
       ::= { sonConfigEntry 4 }

END


src/l2sim/config.h
//-------------------------------------------------------------------------------------------
/** @file DOX_INTERNAL Config.h
 *
 * @brief
 * @author Mindspeed Technologies
 * @version $Revision: 1.96 $
 *
 * COPYRIGHT&copy; 2008-2011 Mindspeed Technologies.
 * ALL RIGHTS RESERVED
 *
 * This is Unpublished Proprietary Source Code of Mindspeed Technologies
 **/
//-------------------------------------------------------------------------------------------

#ifdef __cplusplus
extern "C"
{
#endif

#ifndef _CONFIG_H_
#define _CONFIG_H_

#include "4gmx.h"
#include "hal.h"

#define CPU_NUM  4

#define CPU0_ENABLED

#if CPU_NUM >= 2
#define CPU1_ENABLED
#endif

#if CPU_NUM >= 3
#define CPU2_ENABLED
#endif

#if CPU_NUM >= 4
#define CPU3_ENABLED
#endif

// It will be defined with section name to locate all variables
// with this marker in NCNB section
#define NCNBVAR	__attribute__((section(".NCNBSECT.")))
#define EXT_ICPU_VAR	__attribute__((section(".EXT_ICPU_SECT.")))
#define EMUL_DDR1	__attribute__((section(".EMUL_DDR1.")))
#define VARADDR(x) __attribute__((at(x)))
#define PRV_CORE0           __attribute__((section(".PRV_CORE0.")))

#define MX_MMU_L2_1M_NUMS               100
#define MX_TIMER_FRACTION               10                      // 1ms / MX_TIMER_FRACTION = timer period
#define MX_TIMER_CORES                  (0<<3|0<<2|0<<1|1<<0)   // the bit mask where timer is initialized 
#define MX_WD_CORES                  	(0<<3|0<<2|0<<1|0<<0)   // the bit mask where watchdog is initialized 
#define MX_MMU_INIT_MSG_MASK            (0<<3|0<<2|0<<1|1<<0)
#define MX_APP_INIT_MSG_MASK            (0<<3|0<<2|0<<1|1<<0)

/* The MAP of ARAM-J (stacks and system heap ), the total size is 4 KB */
/* The size of ARAM-J is 8KB, so 4KB for every ARM processor           */
#define SPRV_STACK_SIZE                 (1024 * 100)
#define ABORT_STACK_SIZE                (1024)
#define UNDEF_STACK_SIZE                (1024)
#define FIQ_STACK_SIZE                  (1024 * 10)
#define IRQ_STACK_SIZE                  (1024 * 10)
#define DEF_HEAP_SIZE                   (25 * 1024)
#define RVDS_SYSTEM_HEAP                (50 * 1024)
#define IRAM_HEAP_ADDR                  (IRAM_BASEADDR + 0x10000)
#define IRAM_HEAP_SIZE                  ((256 * 1024) - 0x10000)
#define APP_HEAP_SIZE                   (50 * 1024 * 1024)
#define NCNB_HEAP_SIZE                  (8 * 1024 * 1024)

//******************* [DDR1 parameters] *******************************
//                    -----------------
//   These parameters define DDR-1 heaps, there are two configurations:
//       - DDR-1 is enabled to be used (real DDR-1 HW is used)
//       - DDR-1 is disabled to be used (special compilation of SYSCORE, see: NO_USE_DDR1)
//         in this case DDR-0 is used only and DDR-1 heaps are mapped to DDR-0 space
//*********************************************************************

#ifndef NO_USE_DDR1
    #define DDR1_CACHEABLE_HEAP_SIZE    (128 * 1024 * 1024)  // set to 0 to remove DDR1 cacheable heap
#else
    #define DDR1_CACHEABLE_HEAP_SIZE    (128 * 1024 * 1024)  // set to 0 to remove DDR1 cacheable heap
#endif
//*********************************************************************

#define ETH_NUM_TX_BUFS                 (32)
#define ETH_NUM_RX_BUFS                 (32)

#ifndef NO_USE_DDR1
    #define SVSR_API_LOGGER_BUF_SIZE   (22 * 1024 * 1024)  // set to 0 to remove DDR1 cacheable heap
#else
    #define SVSR_API_LOGGER_BUF_SIZE   (22 * 1024 * 1024)  // set to 0 to remove DDR1 cacheable heap
#endif

//#define UNIDIAG_ENABLED
/*
#define PHY_TXSTART_REQ     1
#define PHY_TXSTART_CONF    2
#define PHY_TXSTART_IND     3
#define PHY_TXSDU_REQ       4
#define PHY_TXSDU_CONF      5
#define PHY_TXEND_IND       6
#define PHY_RXSTART_REQ     7
#define PHY_RXSTART_CONF    8
#define PHY_RXSTART_IND     9
#define PHY_RXSDU_IND       10
#define PHY_RXEND_IND       11
#define PHY_INIT_REQ        12
#define PHY_INIT_IND        13
#define PHY_RXSTATUS_IND    14
#define PHY_RECONFIG_REQ    15
#define PHY_RECONFIG_CNF    16
#define PHY_START_REQ       17
#define PHY_START_CONF      18
#define PHY_STOP_REQ        19
#define PHY_STOP_CONF       20
#define PHY_STOP_IND        21
#define PHY_TXHIADCIUL_REQ  22
#define PHY_TXHISDU_REQ     23
#define PHY_TXDCIULSDU_REQ  24
*/
#define SVSR_API_LOGGER_DEF_MASK        (0)
#define SVSR_API_LOGGER_DEF_MASK_ENA        ((1<<1)|(1<<3)|(1<<4)|(1<<7)|(1<<10)|(1<<12)|(1<<14)|(1<<17)|(1<<19)|(1<<22)|(1<<23)|(1<<24))

#define MLOG_BUF_SIZE                   (MLOG_FRAME_LIMIT * MLOG_FRAME_SIZE * (CPU_NUM + MLOG_DEVICE_COUNT) + 1024*1024)
#define GEN_LOG_BUF_SIZE                (1024)
#define GEN_LOG_OPTIONS                 (LOG_CTRL_RAD_MDMA | LOG_CTRL_SYNCPU)
#define GEN_LOG_HEAP                    (DDR1NcnbHeap)
/** @brief The patterns for stacks filling */

#define SVSR_STACK_PATTER               0xCCCCCCCC
#define ABORT_STACK_PATTER              0xCC
#define UNDEF_STACK_PATTER              0xCC
#define FIQ_STACK_PATTER                0xCC
#define IRQ_STACK_PATTER                0xCC

/** @brief The high and lower application version */

#define APP_H_VER                       0x04
#define APP_L_VER                       0x00

/**********************************************************************
*     The flags of the conditional compilation                        *
***********************************************************************/
//#define ARM_LVECTORS_PROTECTION
//#define SYS_INFO_ENABLED
//#define APP_4GMX_TRACER_ENABLED
#define GIC_INIT
#define LOCAL_TIMER_INIT
#define SYS_TIMER_INIT
#define PRINTF_INIT0
#define PRINTF_INIT1
#define PRINTF_INIT2
//#define DSP_PRINT_INIT
#define CMD_LINE_INIT
//#define TIMER_TEST
#define WIMAX_INIT
#define LTE_INIT
//#define IRAM_FEC_MDMA_ALLOC_ENABLED
#define GIC_IRQ_PROTECTION_ENABLED
#define SVSR_API_LOGGER_ENABLED
#define SPI_DRV_INIT
#define RDMEM_CACHE_CONTROL_ENABLED
#define GENLOG_ENABLED
#define L2_ALWAYS_ENABLED
//#define MMU_CFG_APP_HEAP_AS_64K_PAGES
#define SPACC_DRV_ENABLED
//#define SRIO_DRV_ENABLED
#define REX_ENABLED
#define IRAM_HEAP_INIT
//#define PROFILE_T3K
//#define X7_DEV_ENABLED
//#define PM_ENABLED
//#define DL_IQ_PATTERN_ENABLED
#define REX_S_FORCED_ENABLED
#define UNIDIAG_ENABLED
//#define IQPHONE_ENABLED
#define SWDIAG_EN // enables Turner Software Diagnostics module
//#define HEAPS_PRINT_INFO_ENABLED
//#define MMU_PRINT_INFO_ENABLED

#ifdef MX_TRACER_ENABLED
    #define SYS_TRACER_ENABLED      // To enable tracing partitions/queues
#endif

#ifdef IRAM_FEC_MDMA_ALLOC_ENABLED

#define IRAM_HEAP_INIT
#define MDMA_FEC_HEAP   IramHeap

#else

#define MDMA_FEC_HEAP   DspHeap

#endif

/* uncomment this to compile code for IPHASE board */
//#define IPHASE_36701_BOARD

#define SPI_HEAP            HNULL      // HNULL = the default core heap is used
#define SPI_DEF_DEV_NUM     10

// these macro variables enable/disable corresponding startup tests
#define DIAG_FEC_ENABLED
#define DIAG_FP_ENABLED
//#define DIAG_FP_PRINT_ALWAYS  // to print messages even if FP diagnostics is ok
#define DIAG_DSP_ENABLED
//#define DIAG_DSP_PRINT_ALWAYS  // to print messages even if CEVA diagnostics is ok
#define DIAG_ARM_ENABLED
#define DIAG_SCH_ENABLED
//#define DIAG_I2C_ENABLED
//#define DIAG_SPI_ENABLED
//#define DIAG_PCIE_ENABLED
#ifdef SPACC_DRV_ENABLED
#define DIAG_SPACC_ENABLED
#endif

#define CLINE_INIT
#define PHY_GEM_ENABLED         // To allow linux to enable this device
//#define PHY_GEM_FORCED_CORE0    // GEMAC-1 location, core-0
//#define PHY_GEM_FORCED_CORE1    // GEMAC-1 location, core-1
//#define PHY_GEM_FORCED_CORE2    // GEMAC-1 location, core-2
//#define PHY_GEM_FORCED_CORE3    // GEMAC-1 location, core-3
//#define PHY_GEM_IGNORE_LINUX_ENABLE_GEM
#define SYS_SHUTDOWN_DRV_ENABLED
#define I_CPU_RX_CACHED
#define I_CPU_TX_CACHED
#define SVSR_ON_ARM_1_ENABLED
#define SVSR_ON_ARM_2_ENABLED
#define SVSR_ON_ARM_3_ENABLED
//#define SVSR_INIT_MSG_ENABLED   // to enable initialization messages (just message not error messages)
#define PROC_ABORTS_ENABLED
#define SYSCORE_DEFAULT_UART_ID  1 // Console manager works with: 0:UART-0(not supported in current configuration), 1:UART-1, 2:UART-2, 3:disabled
//#define UART_DISABLE_FORCED // disables UART and console using
#define CMGR_TURNED_OFF 3

#define MAX_NUM_ANT_SUPPORTED           2
#define CPRI_DMA_BUF_ADDR               0xFC200000
#define CPRI_DMA_BUF_SIZE               (2*4*61440)
#define CPRI_RXDMA_BUF_ADDR             (CPRI_DMA_BUF_ADDR + CPRI_DMA_BUF_SIZE)
#define CPRI_RXDMA_BUF_SIZE              CPRI_DMA_BUF_SIZE

/**********************************************************************
*     Use this flag to enable/disable HBI interface and               *
*     enable/disable API routing to/from the host                     *
***********************************************************************/

#if defined(TIMER_TEST)&&!defined(GIC_INIT)
#error "Please define <GIC_INIT> macro"
#endif

typedef struct _STACKS_HEAP_
{
    char spvs[SPRV_STACK_SIZE];
    char abort[ABORT_STACK_SIZE];
    char undef[UNDEF_STACK_SIZE];
    char fiq[FIQ_STACK_SIZE];
    char irq[IRQ_STACK_SIZE];

} STACKSHEAP;

extern STACKSHEAP stacks_cpu0;
extern STACKSHEAP stacks_cpu1;
extern STACKSHEAP stacks_cpu2;
extern STACKSHEAP stacks_cpu3;

extern const UINT32 mmutable[][4096];
extern const UINT32 mmu_l2_table[CPU_NUM][MX_MMU_L2_1M_NUMS * 256];

extern char defheapbuf[][DEF_HEAP_SIZE];
extern MXCONTEXT MxCtx[];
extern MXSMPCTX  MxSmpCtx;

#define STACK_BASE_ADDR(cpu)  ( &stacks_cpu0 + (cpu))

#define DEFHEAP_BASE_ADDR(cpu)  (defheapbuf [(cpu)])

#define STACK_BASE_ADDR_CPU0  (stacks_cpu0)
#define STACK_BASE_ADDR_CPU1  (stacks_cpu1)
#define STACK_BASE_ADDR_CPU2  (stacks_cpu2)
#define STACK_BASE_ADDR_CPU3  (stacks_cpu3)

#define PRI_DDR_MON         0x1
#define PRI_DSP             0x1
#define PRI_EVMFPGA         0x0     // Highest
#define PRI_GEM             0x1
#define PRI_HBI             0x1
#define PRI_PCIE            0x1
#define PRI_RDMEM           0x1
#define PRI_SCH_TASK        0x1
#define PRI_SPI             0x1
#define PRI_SRIO            0x1
#define PRI_SYS_SHTDWN      0x1
#define PRI_SYS_TIMERS      0x1
#define PRI_TIMER           0x0     // Highest
#define PRI_CPRI            0x0     // Highest
#define PRI_DL_FEC          0x1
#define PRI_UL_FEC          0x1
#define PRI_MDMA_SYS        0x1
#define PRI_MDMA_RAD        0x1
#define PRI_MDMA_SPU        0x1
#define PRI_MAP_ARM         0x1
#define PRI_MAP_DMA         0x1
#define PRI_UART            0x1
#define PRI_ICPU            0x1
#define PRI_PMON            0x1
#define PRI_REX             0x1
#define PRI_REX_SERV        0x1
#define PRI_SPAC            0x1


extern void * _memcpy (void *dest, const void *src, unsigned int count);

#ifdef __CC_ARM
#define CPU_IDLE() __nop ()
#else
#define CPU_IDLE()
#endif

#endif /*_CONFIG_H_*/

#ifdef __cplusplus
}
#endif

